Use of deprecated functions seem to break compilation with recent kernels

As reported by alan_ and kaliohelix on discord - 
```pci_alloc_consistent``` , ```pci_free_consistent``` and possibly more functions currently used have been removed

See e.g:

https://lore.kernel.org/lkml/20220106222804.GA330366@bhelgaas/T/

Looks like it's mostly just replacing with different functions with potentially slightly different signature that is needed:
https://www.mail-archive.com/linux-kernel@vger.kernel.org/msg2397228.html
Seems this has been around for a bit

Building against Linux 6.1:

```
make[1]: Entering directory '/src/linux/linux/work/v6.1'
  CC [M]  /src/linux/linux/work/cxadc-module/cxadc.o
/src/linux/linux/work/cxadc-module/cxadc.c: In function ‘alloc_risc_inst_buffer’:
/src/linux/linux/work/cxadc-module/cxadc.c:221:31: error: implicit declaration of function ‘pci_alloc_consistent’ [-Werror=implicit-function-declaration]
  221 |         ctd->risc_inst_virt = pci_alloc_consistent(ctd->pci, ctd->risc_inst_buff_size, &ctd->risc_inst_phy);
      |                               ^~~~~~~~~~~~~~~~~~~~
/src/linux/linux/work/cxadc-module/cxadc.c:221:29: warning: assignment to ‘unsigned int *’ from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
  221 |         ctd->risc_inst_virt = pci_alloc_consistent(ctd->pci, ctd->risc_inst_buff_size, &ctd->risc_inst_phy);
      |                             ^
/src/linux/linux/work/cxadc-module/cxadc.c: In function ‘free_risc_inst_buffer’:
/src/linux/linux/work/cxadc-module/cxadc.c:235:17: error: implicit declaration of function ‘pci_free_consistent’ [-Werror=implicit-function-declaration]
  235 |                 pci_free_consistent(ctd->pci, ctd->risc_inst_buff_size, ctd->risc_inst_virt, ctd->risc_inst_phy);
      |                 ^~~~~~~~~~~~~~~~~~~
cc1: some warnings being treated as errors
make[2]: *** [scripts/Makefile.build:250: /src/linux/linux/work/cxadc-module/cxadc.o] Error 1
make[1]: *** [Makefile:1992: /src/linux/linux/work/cxadc-module] Error 2
make[1]: Leaving directory '/src/linux/linux/work/v6.1'
```

--- a/cxadc.c
+++ b/cxadc.c
@@ -38,6 +38,7 @@
 #if LINUX_VERSION_CODE >= KERNEL_VERSION(4, 21, 0)
 #define dma_zalloc_coherent dma_alloc_coherent
 #endif
+
 static int latency = -1;
 static int audsel = -1;
 static int vmux = 2;
@@ -218,7 +219,7 @@ static int alloc_risc_inst_buffer(struct cxadc *ctd)
 {
 	/* add 1 page for sync instruct and jump */
 	ctd->risc_inst_buff_size = (VBI_DMA_BUFF_SIZE/CLUSTER_BUFFER_SIZE)*8+PAGE_SIZE;
-	ctd->risc_inst_virt = pci_alloc_consistent(ctd->pci, ctd->risc_inst_buff_size, &ctd->risc_inst_phy);
+	ctd->risc_inst_virt = dma_alloc_coherent(&ctd->pci->dev, ctd->risc_inst_buff_size, &ctd->risc_inst_phy, GFP_KERNEL);
 	if (ctd->risc_inst_virt == NULL)
 		return -ENOMEM;
 	memset(ctd->risc_inst_virt, 0, ctd->risc_inst_buff_size);
@@ -232,7 +233,7 @@ static int alloc_risc_inst_buffer(struct cxadc *ctd)
 static void free_risc_inst_buffer(struct cxadc *ctd)
 {
 	if (ctd->risc_inst_virt != NULL)
-		pci_free_consistent(ctd->pci, ctd->risc_inst_buff_size, ctd->risc_inst_virt, ctd->risc_inst_phy);
+		dma_free_coherent(&ctd->pci->dev, ctd->risc_inst_buff_size, ctd->risc_inst_virt, ctd->risc_inst_phy);
 }
 
 static int make_risc_instructions(struct cxadc *ctd)
@@ -550,7 +551,7 @@ static int cxadc_probe(struct pci_dev *pci_dev,
 		return -EBUSY;
 	}
 
-	ctd = kmalloc(sizeof(*ctd), GFP_ATOMIC);
+	ctd = kmalloc(sizeof(*ctd), GFP_KERNEL);
 	if (!ctd) {
 		rc = -ENOMEM;
 		dev_err(&pci_dev->dev, "cxadc: kmalloc failed\n");
@@ -580,7 +581,7 @@ static int cxadc_probe(struct pci_dev *pci_dev,
 	for (i = 0; i < MAX_DMA_PAGE; i++) {
 		dma_addr_t dma_handle;
 
-		ctd->pgvec_virt[i] = (void *)dma_zalloc_coherent(&ctd->pci->dev, 4096, &dma_handle, GFP_KERNEL);
+		ctd->pgvec_virt[i] = dma_zalloc_coherent(&ctd->pci->dev, 4096, &dma_handle, GFP_KERNEL);
 		if (ctd->pgvec_virt[i] != 0) {
 			ctd->pgvec_phy[i] = dma_handle;
 			total_size += pgsize;

